pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +435,64 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
 	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
 	return err
 }
+
+func (m *MonitorManager) smoothMigrationToStatefulSet(monitor *v1alpha1.TidbMonitor) error {
+	exist, err := m.deps.TypedControl.Exist(client.ObjectKey{
+		Namespace: monitor.Namespace,
+		Name:      monitor.Name,
+	}, &appsv1.Deployment{})
+	if err != nil {
+		klog.Errorf("tm[%s/%s]'s deployment failed to get,err: %v", monitor.Namespace, monitor.Name, err)
+		return err
+	}
+	if exist {
+		err = m.deps.TypedControl.Delete(monitor, &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      monitor.Name,
+				Namespace: monitor.Namespace,
+			},
+		})
+		if err != nil {
+			klog.Errorf("tm[%s/%s]'s deployment failed to delete,err: %v", monitor.Namespace, monitor.Name, err)
+			return err
+		}
+		return nil
+	} else {
+		stsPvcName := fmt.Sprintf("%s-0", GetMonitorObjectName(monitor))
+		stsPvc := getMonitorPVC(stsPvcName, monitor)
+		stsPvc, err := m.deps.TypedControl.CreateOrUpdatePVC(monitor, stsPvc, false)
+
+		if err != nil {
+			klog.Errorf("tm[%s/%s]'s pvc failed to sync,err: %v", monitor.Namespace, monitor.Name, err)
+			return err
+		}
+
+		deploymentPvcName := GetMonitorObjectName(monitor)
+		deploymentPvc, err := m.deps.PVCLister.PersistentVolumeClaims(monitor.Namespace).Get(deploymentPvcName)
+		if err != nil {
+			return err
+		}
+		if deploymentPvc != nil {
+			// update meta info for pv
+			deploymentPv, err := m.deps.PVLister.Get(deploymentPvc.Spec.VolumeName)
+			if err != nil {
+				return err
+			}
+			deploymentPv.Spec.ClaimRef.Name = stsPvc.Name
+			_, err = m.deps.PVControl.UpdateMetaInfo(monitor, deploymentPv)
```

Comment:
It's not enough to update the meta info, we have to bind the PV to the new PVC of sts, is it easy to do?

Reply:
I think we just need to update claimRef .